oauth2client/contrib/gce.py

```diff
@@ -48,28 +45,67 @@
 """
 
 
-def _get_service_account_email(http_request=None):
-    """Get the GCE service account email from the current environment.
-
+def _get_metadata(http_request=None, path=None, recursive=True):
```

Comment:
So this is a bigger design question: Lazily populate values, or populate them in the constructor. 

I went with lazy population because that is what the original author did, and also, there is a chance that a user will never use the some properties, resulting in fewer overall calls. 

I have cached the values, as well as consolidating calls (e.g. scopes and email are fetched together)  as much as possible. If you see other opportunities to cache them please let me know.